test/suite-functional/api/functional.thenable.spec.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+declare (strict_types = 1); // @codeCoverageIgnore
+
+namespace Recoil;
+
+use Eloquent\Phony\Phony;
+
+context('can be invoked by yielding a thenable', function () {
```

Comment:
What about the "fulfil" vs "resolved"? Different names seem to be used in different libraries, which one do you want in recoil?